File: utils/Signal.py
from utils import Peak


class Signal:
    def __init__(self,
                 length: int,
                 peak_n: int,
                 peaks: list[Peak],
                 data: list[int]):
        self.length = length
        self.peak_n = peak_n
        self.peaks = peaks
        self.data = data


class SignalLight:
    def __init__(self,
                 length: int,
                 peaks: list[float],
                 data: list[int]):
        self.length = length
        self.peaks = peaks
        self.data = data

File: src/cpp_source/Signal.cpp
#include "Signal.h"
#include <initializer_list>

#define MAX_PEAK_NUMBER 3
#define SIGNAL_LENGTH 1024


Signal::Signal(std::vector<float>& peaks, std::array<int, SIGNAL_LENGTH>& data)
{
    m_peaks = peaks;
    std::copy(std::begin(data), std::end(data), std::begin(m_data));

    std::fill(std::begin(m_target), std::end(m_target), 0);

    for (float peak : peaks) {
        m_target[int(peak)] = 1;
    }
};